### ISSUE TITLE
simtk.openmm.OpenMMException: Particle coordinate is nan

### ISSUE THREAD

**Issue by DrJesseHansen** [deepmind/alphafold#741]
hello. We are struggling endlessly to get 2.3.1 running, it would be great if someone could help us with these issues.  Here is the latest.  This is running on A100.  

Here is the submission script:

```
jhansen@bea81:~/AF2_testrun/multimer/gpu228$ cat AF2_multimer.job 
#!/bin/bash

#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --cpus-per-task=8
#SBATCH --mem=200GB

#SBATCH --time=48:00:00
#SBATCH --no-requeue

#SBATCH --partition=gpu
#SBATCH --gres=gpu:1
MY_PROTEIN_PATH=test_multi.fasta

echo $HOSTNAME

module purge
module load alphafold/2.3.1

export OPENMM_CUDA_COMPILER=$(which nvcc)

python3 /mnt/nfs/clustersw/Debian/bullseye/cuda/11.2/alphafold/2.3.1/alphafold-2.3.1/run_alphafold.py \
	--model_preset=multimer \
	--fasta_paths=$MY_PROTEIN_PATH \
	--output_dir=$(dirname $MY_PROTEIN_PATH) \
	--data_dir=/nfs/scistore14/rcsb/alphafold.databases/ \
	--mgnify_database_path=/nfs/scistore14/rcsb/alphafold.databases/mgnify/mgy_clusters_2022_05.fa \
	--template_mmcif_dir=/nfs/scistore14/rcsb/alphafold.databases/pdb_mmcif/mmcif_files/ \
	--max_template_date=2023-03-01 \
	--obsolete_pdbs_path=/nfs/scistore14/rcsb/alphafold.databases/pdb_mmcif/obsolete.dat \
	--use_gpu_relax=true \
	--bfd_database_path=/nfs/scistore14/rcsb/alphafold.databases/bfd/bfd_metaclust_clu_complete_id30_c90_final_seq.sorted_opt \
        --uniref30_database_path=/nfs/scistore14/rcsb/alphafold.databases/uniref30/UniRef30_2021_03 \
	--uniref90_database_path=/nfs/scistore14/rcsb/alphafold.databases/uniref90/uniref90.fasta \
	--pdb_seqres_database_path=/nfs/scistore14/rcsb/alphafold.databases/pdb_seqres/pdb_seqres.txt \
	--uniprot_database_path=/nfs/scistore14/rcsb/alphafold.databases/uniprot/uniprot.fasta
```

here is the failed output:

```
Traceback (most recent call last):
  File "/mnt/nfs/clustersw/Debian/bullseye/cuda/11.2/alphafold/2.3.1/alphafold-2.3.1/run_alphafold.py", line 432, in <module>
    app.run(main)
  File "/mnt/nfs/clustersw/Debian/bullseye/cuda/11.2/alphafold/2.3.1/lib/python3.9/site-packages/absl/app.py", line 312, in run
    _run_main(main, args)
  File "/mnt/nfs/clustersw/Debian/bullseye/cuda/11.2/alphafold/2.3.1/lib/python3.9/site-packages/absl/app.py", line 258, in _run_main
    sys.exit(main(argv))
  File "/mnt/nfs/clustersw/Debian/bullseye/cuda/11.2/alphafold/2.3.1/alphafold-2.3.1/run_alphafold.py", line 408, in main
    predict_structure(
  File "/mnt/nfs/clustersw/Debian/bullseye/cuda/11.2/alphafold/2.3.1/alphafold-2.3.1/run_alphafold.py", line 243, in predict_structure
    relaxed_pdb_str, _, violations = amber_relaxer.process(
  File "/nfs/scistore07/clustersw/debian/bullseye/cuda11.2/alphafold/2.3.1/alphafold-2.3.1/alphafold/relax/relax.py", line 62, in process
    out = amber_minimize.run_pipeline(
  File "/nfs/scistore07/clustersw/debian/bullseye/cuda11.2/alphafold/2.3.1/alphafold-2.3.1/alphafold/relax/amber_minimize.py", line 468, in run_pipeline
    pdb_string = clean_protein(prot, checks=checks)
  File "/nfs/scistore07/clustersw/debian/bullseye/cuda11.2/alphafold/2.3.1/alphafold-2.3.1/alphafold/relax/amber_minimize.py", line 173, in clean_protein
    fixed_pdb = cleanup.fix_pdb(pdb_file, alterations_info)
  File "/nfs/scistore07/clustersw/debian/bullseye/cuda11.2/alphafold/2.3.1/alphafold-2.3.1/alphafold/relax/cleanup.py", line 55, in fix_pdb
    fixer.addMissingAtoms(seed=0)
  File "/mnt/nfs/clustersw/Debian/bullseye/cuda/11.2/alphafold/2.3.1/lib/python3.9/site-packages/pdbfixer-1.8.1-py3.9.egg/pdbfixer/pdbfixer.py", line 954, in addMissingAtoms
    mm.LocalEnergyMinimizer.minimize(context)
  File "/mnt/nfs/clustersw/Debian/bullseye/cuda/11.2/alphafold/2.3.1/lib/python3.9/site-packages/simtk/openmm/openmm.py", line 18468, in minimize
    return _openmm.LocalEnergyMinimizer_minimize(context, tolerance, maxIterations)
simtk.openmm.OpenMMException: Particle coordinate is nan
```

**Comment by smturzo** [comment]
Does AF231 work if you run it without amber relax? As in 
--use_gpu_relax=false.
Also reading the GH readme I couldn't find this particular flag "--use_gpu_relax" you mention. I found these instead:
"--enable_gpu_relax=false"
 "--models_to_relax=none"

Maybe try running AF231 without Amber Relax part and see if you get structures. If you do so, visualize the structures. If they look really wild then Amber relax is probably failing due to a high number of clashes and/or bad geometries (this could possibly explain the NotaNumber(nan) issue. I have had that happen for a really large multimeric complex.

**Comment by DrJesseHansen** [comment]
thank you for your response.  

Indeed I do only see --enable_gpu_relax on the GH.   However this must be a mistake (and should be fixed to avoid confusion).  Hopefully the repo owners see this. 

After loading AF2.3.1 and running it with that flag I get the following error:
```
FATAL Flags parsing error: Unknown command line flag 'enable_gpu_relax'. Did you mean: use_gpu_relax ?
Pass --helpshort or --helpfull to see help on flags.
```

Further, when I invoked the helpful through terminal I saw this in the output:
`--[no]use_gpu_relax: Whether to relax on GPU. Relax on GPU can be much faster than CPU, so it is recommended to enable if possible. GPUs must be available if this setting is enabled.`

So for now I will leave it as --use_gpu_relax.    However, as you suggested I will try turning it off to see if that fixes the issue.  Thank you for the suggestion!  As a side note, this error is happening even on small multimers and for different input sequences.

**Comment by Htomlinson14** [comment]
Hi thanks for raising. This should be fixed in v2.3.2. Please raise again if this persists!

**Comment by DrJesseHansen** [comment]
> Hi thanks for raising. This should be fixed in v2.3.2. Please raise again if this persists!

thank you.  Is there a solution for now?

**Comment by Htomlinson14** [comment]
To clarify - v2.3.2 has been released
